src/main/java/com/minecolonies/structures/fake/FakeWorld.java

```diff
@@ -60,16 +101,41 @@ protected boolean isChunkLoaded(final int x, final int z, final boolean allowEmp
         return false;
     }
 
+    @Override
+    public boolean setBlockState(final BlockPos pos, final IBlockState state)
+    {
+        stateHashMap.put(pos, state);
+        return true;
+    }
+
+    @Override
+    public void setTileEntity(final BlockPos pos, @Nullable final TileEntity tileEntityIn)
+    {
+        entityHashMap.put(pos, tileEntityIn);
+    }
+
     @NotNull
     @Override
     public IBlockState getBlockState(final BlockPos pos)
     {
+        if(simulateWorld)
+        {
+            if(stateHashMap.containsKey(pos))
+            {
+                return stateHashMap.get(pos);
+            }
+            return Blocks.AIR.getDefaultState();
```

Comment:
This is not right.
Use a reference to the world we are rendering in here and then grab the data from there.
Makes it so that fences also connect to already existing fences.